crates/bevy_pbr/src/fog.rs

```diff
@@ -0,0 +1,223 @@
+use crate::ReflectComponent;
+use bevy_ecs::{prelude::*, query::QueryItem};
+use bevy_reflect::Reflect;
+use bevy_render::{color::Color, extract_component::ExtractComponent, prelude::Camera};
+
+/// Configures the “classic” computer graphics [distance fog](https://en.wikipedia.org/wiki/Distance_fog) effect,
+/// in which objects appear progressively more covered in atmospheric haze as they move further away from the camera.
+/// Affects meshes rendered via the PBR [`StandardMaterial`](crate::StandardMaterial).
+///
+/// ## Falloff
+///
+/// The rate at which fog intensity increases with distance is controlled by the falloff mode.
+/// Currently, the following fog falloff modes are supported:
+///
+/// - [`FogFalloff::Linear`]
+/// - [`FogFalloff::Exponential`]
+/// - [`FogFalloff::ExponentialSquared`]
+/// - [`FogFalloff::Atmospheric`]
+///
+/// ## Example
+///
+/// ```
+/// # use bevy_ecs::prelude::*;
+/// # use bevy_render::prelude::*;
+/// # use bevy_core_pipeline::prelude::*;
+/// # use bevy_pbr::prelude::*;
+/// # fn system(mut commands: Commands) {
+/// commands.spawn((
+///     // Setup your camera as usual
+///     Camera3dBundle {
+///         // ... camera options
+/// #       ..Default::default()
+///     },
+///     // Add fog to the same entity
+///     FogSettings {
+///         color: Color::WHITE,
+///         falloff: FogFalloff::Exponential { density: 1e-3 },
+///         ..Default::default()
+///     },
+/// ));
+/// # }
+/// # bevy_ecs::system::assert_is_system(system);
+/// ```
+///
+/// ## Material Override
+///
+/// Once enabled for a specific camera, the fog effect can also be disabled for individual
+/// [`StandardMaterial`](crate::StandardMaterial) instances via the `no_fog` flag.
+#[derive(Debug, Clone, Component, Reflect)]
+#[reflect(Component)]
+pub struct FogSettings {
+    /// The color of the fog effect.
+    ///
+    /// **Tip:** The alpha channel of the color can be used to “modulate” the fog effect without
+    /// changing the fog falloff mode or parameters.
+    pub color: Color,
+
+    /// The color used for the fog the view direction aligns with directional lights
+    /// Produces a “halo” or light dispersion effect (e.g. around the sun)
+    pub scattering_color: Color,
```

Comment:
> I think this variable name could be improved. The best I can come up with right now is `directional_light_color`.

👍 Renamed it to that, also renamed `scattering_exponent` to `directional_light_exponent` for consistency.

> That makes me think that now we support multiple directional lights, we could apply this for each light source and use the light's colour instead of setting it here... :)

Modified it so that it takes the light color into account, but also multiplies it with this value, to preserve some degree of artistic control. (e.g. in case you want the atmosphere glow form the sun to look purple-ish but don't want your directional light to be _actually_ purple)